### PR TITLE
API-2.5: documentation and specs

### PR DESCRIPTION
**Pull request by wvengen**
As surfaced in #576: add documentation and specs for basic endpoints, as well as a fix for Doorkeeper 5.
Will add specs in #572 and #573.

### CODE REVIEW

File: doc/swagger.v1.yml [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+  /config:
+    get:
+      summary: configuration variables
+      tags:
+        - 7. General
+      responses:
+        200:
+          description: success
+          schema:
+            type: object
```
Comment (wvengen) [inline doc/swagger.v1.yml]: This is pretty general right now. Do you think it makes sense to include all config keys here? It's a little work, and the configuration list in the backend is much more extensive. Or does a rake task that copies config key descriptions here make sense at some point ... ?

File: doc/swagger.v1.yml [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+  /navigation:
+    get:
+      summary: navigation
+      tags:
+        - 7. General
+      responses:
+        200:
+          description: success
+          schema:
+            type: object
```
Comment (wvengen) [inline doc/swagger.v1.yml]: I guess it would be nice to specify this in the definitions, still.

File: doc/swagger.v1.yml (hunk 1 above)
Comment (paroga) [reply doc/swagger.v1.yml]: do we have a full list of all keys? Doesn't it depend on the activated plugins?

File: doc/swagger.v1.yml (hunk 2 above)
Comment (paroga) [reply doc/swagger.v1.yml]: same as with the previous point.

Comment (wvengen) [reply doc/swagger.v1.yml]: Added this, the form is expected to remain the same (array of objects with name+url or name+items).

File: doc/swagger.v1.yml (hunk 1 above)
Comment (wvengen) [reply doc/swagger.v1.yml]: That's a good point, actually.

File: db/migrate/20181013195028_add_confidential_to_applications.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
```
Comment (paroga) [inline db/migrate/20181013195028_add_confidential_to_applications.rb]: do we really need this here?

File: db/migrate/20181013195028_add_confidential_to_applications.rb [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+class AddConfidentialToApplications < ActiveRecord::Migration
+  def change
+    add_column(
```
Comment (paroga) [inline db/migrate/20181013195028_add_confidential_to_applications.rb]: can you use the style of the other migration files here too? (one line, no brackets)

File: spec/support/api_helper.rb [hunk 5]
```diff
@@ -0,0 +1,18 @@
+module ApiHelper
+  extend ActiveSupport::Concern
+
+  included do
+    let(:user) { create :user }
+    let(:access_token) { create(:oauth2_access_token, resource_owner_id: user.id).token }
+    let(:authorization) { "Bearer #{token}" }
```
Comment (paroga) [inline spec/support/api_helper.rb]: is this used? `token` vs `access_token`?

File: spec/support/api_helper.rb [hunk 6]
```diff
@@ -2,17 +2,17 @@ module ApiHelper
   extend ActiveSupport::Concern
 
   included do
-    let(:user) { create :user }
-    let(:access_token) { create(:oauth2_access_token, resource_owner_id: user.id).token }
-    let(:authorization) { "Bearer #{token}" }
+    let(:user) { create(:user) }
+    let(:api_access_token) { create(:oauth2_access_token, resource_owner_id: user.id).token }
```
Comment (paroga) [inline spec/support/api_helper.rb]: why not do that already correct in the previous commit?

File: db/migrate/20181013195028_add_confidential_to_applications.rb (hunk 3 above)
Comment (wvengen) [reply db/migrate/20181013195028_add_confidential_to_applications.rb]: This migration was generated by the doorkeeper upgrade procedure, from [this template](https://github.com/doorkeeper-gem/doorkeeper/blob/3b61b6c3e99a309c8e5e25f86ee6ac2c8131b05d/lib/generators/doorkeeper/templates/add_confidential_to_applications.rb.erb). I agree that we'd better use our own style here.

**Comment by wvengen** [comment]
After addressing the issues, I think it's ok to merge this, and intend to do so after a week.

**Pull request merged**